pontoon/base/models/project.py

```diff
@@ -276,11 +277,24 @@ def checkout_path(self):
     def get_latest_activity(self, locale=None):
         from pontoon.base.models.project_locale import ProjectLocale
 
-        return ProjectLocale.get_latest_activity(self, locale)
+        if locale is None:
+            return (
+                self.latest_translation.latest_activity
+                if self.latest_translation
+                else None
+            )
 
-    @property
-    def avg_string_count(self):
-        return int(self.total_strings / self.enabled_locales)
+        pl_fetched = getattr(self, "fetched_project_locale", None)
+        project_locale = (
+            pl_fetched[0]
+            if pl_fetched
+            else ProjectLocale.objects.filter(project=self, locale=locale).first()
```

Comment:
I think this should be sufficient (and save us some more DB queries)? We only get here if the locale is not available for the project, in which case we don't have latest activity data anyways.

```suggestion
            else None
```

Reply:
If the above logic "works", you might be able to simplify the code a little (didn't actually test it):

```
pl_fetched = getattr(self, "fetched_project_locale", None)
if pl_fetched:
    project_locale = pl_fetched[0]
    return getattr(project_locale.latest_translation, "latest_activity", None)
return None
```

Reply:
Ah, true, because this is only called from `ajax_projects()`.